documentation/role-requirements.md

```diff
@@ -0,0 +1,141 @@
+    vars:
+      firewall:
+        - port: 9876
+          state: enabled
+```
+
+The role2collection script will also have to change the README, and anywhere
+else external roles are referenced.  The role2collection script supports the
+`--extra-mapping` parameter.  Maybe we could add something like
+`--extra-mapping-from /path/to/meta/role-requirements.yml`.  The role2collection
+script would read each role and convert each one like
+`--extra-mapping org.rolename:namespace.collection.rolename`.
+
+The role2collection script will not need to do anything with
+`meta/role-requirements.yml` - it will not need to add these to the collection
+`galaxy.yml` as dependencies.  Any dependencies on linux-system-roles roles will
+be automatically satisfied by the collection, since the collection includes all
+of them.  Any dependencies on external roles are avoided for now, and left to
+some future developer to figure out.
+
+# Tools/CI
+
+We will need to add support for this to `tox-lsr`.  For example, if I am a
+logging role developer, and I run a logging role test using `tox -e qemu ...`,
+tox-lsr must ensure the certificate, selinux, and firewall roles are available,
+using `role-requirements.yml`, similar to how it handles collection dependencies
+https://github.com/linux-system-roles/tox-lsr/blob/main/src/tox_lsr/test_scripts/runqemu.py#L863
+
+If testing the role by converting it to collection format first, the
+dependencies will also need to be converted into the same collection.  For
+example, if I use `tox -e collection`, and the role has dependencies on system
+roles `roleA` and `roleB`, then the converted collection must have
+`fedora.linux_system_roles.roleA` and `fedora.linux_system_roles.roleB`.
+
+Developers should have the option to specify a local directory.  For example, if
+I have all of the roles checked out under `$HOME/linux-system-roles`, I should
+be able to use `$HOME/linux-system-roles/certificate` etc. when working on the
+logging role (which is especially handy if I am working on both roles in
+conjunction).  Perhaps we can add symlinks under `tests/roles/`, or the user can
+specify something like `tox -e qemu ... --roles-path $HOME/linux-system-roles`
+(but the latter would require that the developer has
+`$HOME/linux-system-roles/linux-system-roles.certificate` i.e. the name of the
+directory must be the FQRN).
+
+We will need to add support for integration test CI.  The test will need to
+install the roles using `role-requirements.yml`, similar to how we install
+collection dependencies.
+
+# RPM/Packaging
+
+We may need to improve support for this in the RPM spec file.  For example, we
+currently do a `sed` of all `linux-system-roles.` strings to `rhel-system-roles.`.
+We need to ensure this works for the role references mentioned above.  We may
+need to use a semantic scanner that understands the YAML structure/Ansible
+syntax, similar to how the role2collection script works, if `sed` is not suitable.
```

Comment:
Would it be a problem if in the downstream legacy role there remained a reference like `linux-system-roles.selinux`, not converted to `rhel-system-roles.selinux`? it would be ugly from the consistency perspective, but would it cause real problems?

Reply:
> Would it be a problem if in the downstream legacy role there remained a reference like `linux-system-roles.selinux`, not converted to `rhel-system-roles.selinux`? it would be ugly from the consistency perspective, but would it cause real problems?

It shouldn't be a problem since the downstream package has both `linux-system-roles.selinux` and `rhel-system-roles.selinux`.  And I don't think users will be poking around in the source code for the role and wondering why there is a reference to `linux-system-roles`.

My main objection to using the legacy roles is that we have to create and maintain a dependency mechanism specifically for them.  I would rather avoid that.